src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -175,6 +176,7 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
 
             builder.ConfigureServices((context, services) =>
             {
+                services.AddSingleton<IHostedService, FunctionAppValidationService>();
```

Comment:
Do we need this service to be running in placeholder mode? You can register this only after specialization.

```
if (!SystemEnvironment.Instance.IsPlaceholderModeEnabled())
{
    services.AddHostedService<FunctionAppValidationService>();
}
```